### PR TITLE
Generify Location on the type of extent

### PR DESCRIPTION
**Pull request by DDoS**
See issue #788.

### CODE REVIEW

**Comment by DDoS** [comment]
These changes are very sensitive to conflicts, so can I get a fast review and approval/rejection?

**Comment by zml2008** [comment]
I agree with this change

**Comment by Zidane** [comment]
Me as well.

File: src/main/java/org/spongepowered/api/GameRegistry.java [hunk 1]
```diff
@@ -503,7 +503,7 @@
      * @return A new invalid identity transform
      * @see #createTransform(Extent)
      */
-    Transform createTransform();
+    <E extends Extent> Transform<E> createTransform();
```
Comment (gratimax) [inline src/main/java/org/spongepowered/api/GameRegistry.java]: Commenting a bit too late, but this could just be `Transform<? extends Extent>`

Comment (gabizou) [inline src/main/java/org/spongepowered/api/GameRegistry.java]: We'd want the Transform to be generified, so you could do `gameRegistry.<World>createTransform()`

**Pull request merged**